onnxscript/rewriter/generic_pattern_test.py

```diff
@@ -246,6 +247,40 @@ def get_rotary_model(self):
         )
         return model
 
+    def test_shared_root_value_test(self):
+        def match_pattern(op, x):
+            t1 = op.Sin(x)
+            t2 = op.Cos(x)
+            return t1, t2
+
+        def apply_pattern(op, x, **_):
+            return op.SinCos(x, domain="com.microsoft", outputs=2)
+
+        rule = pattern.RewriteRule(
+            match_pattern,
+            apply_pattern,
+            matcher=generic_pattern.GenericPatternMatcher,
+        )
+        model_proto = onnx.parser.parse_model(
+            """
+            <ir_version: 7, opset_import: [ "" : 17]>
+            agraph (float[N] y) => (float[N] z)
+            {
+                temp1 = Sin(y)
+                temp2 = Cos(y)
+                z = Add(temp1, temp2)
+            }
+        """
+        )
```

Comment:
Currently we have three ways to build test models: onnx helper, text repr, and ONNX Script. I think it would be good if we consolidate to use one, preferably ONNX Script.

Reply:
I think for unit tests people are using whatever is faster for them. For this particular case, using something else than onnxscript may be better. You need to make sure that the optimized graph is exactly what you have in mind. When you write something like ``op.Split(x, num_outputs=2, axis=-1)``, you need to make sure there are two outputs. onnxscript may have a bug plus you can choose the node and results names more easily.

Reply:
I personally find both the text repr and onnxscript equally compact and readable. But, as Xavier mentions, with the text repr we know exactly what the test-model is (onnxscript could do some expansion, etc.).

Anyway, hope this is not a blocking comment. The PR is ready to be merged with an approval

Reply:
Sg. Thanks!